pkg/landscaper/installations/executions/template/gotemplate.go

```diff
@@ -480,3 +483,62 @@ func resolveComponents(defaultCD *cdv2.ComponentDescriptor, list *cdv2.Component
 
 	return components, nil
 }
+
+func getImageVectorGoFunc(cd *cdv2.ComponentDescriptor, list *cdv2.ComponentDescriptorList) func(args ...interface{}) map[string]interface{} {
+	return func(args ...interface{}) map[string]interface{} {
+		internalCd := cd
+		internalComponents := list
+
+		if len(args) > 2 {
+			panic("Too many arguments for getImageVectorGoFunc")
+		}
+
+		if len(args) >= 1 {
+			data, err := json.Marshal(args[0])
+			if err != nil {
+				panic("Unable to marshal first argument to getImageVectorGoFunc")
+			}
+
+			internalCd = &cdv2.ComponentDescriptor{}
+			if err = codec.Decode(data, internalCd); err != nil {
+				panic("Unable to marshal first argument to component descriptor.")
+			}
+		}
+
+		if len(args) == 2 {
+			componentsData, err := json.Marshal(args[1])
+			if err != nil {
+				panic("Unable to marshal second argument to component descriptor list.")
+			}
+
+			internalComponents = &cdv2.ComponentDescriptorList{}
+			if err := codec.Decode(componentsData, internalComponents); err != nil {
+				panic("Unable to marshal second argument to component descriptor list.")
```

Comment:
```suggestion
				panic("Unable to decode second argument to component descriptor list.")
```
alterntativly unmarshal